ccmlib/common.py

```diff
@@ -76,6 +77,36 @@ class UnavailableSocketError(CCMError):
     pass
 
 
+class LogPatternToVersion(object):
+    def __init__(self, patterns_to_versions):
+        assert type(patterns_to_versions) is dict, "LogPatternToVersion takes a dictionary."
+        self.versions_to_patterns = patterns_to_versions
+
+    def __call__(self, version):
+        pattern = None
+        for key in self.versions_to_patterns.keys():
+                if version >= key:
+                    pattern = self.versions_to_patterns[key]
+        return pattern
+
+    def __iter__(self):
+        return iter(self.versions_to_patterns)
+
+    def __len__(self):
+        return len(self.versions_to_patterns)
+
+    def __repr__(self):
+        return str(self.versions_to_patterns)
```

Comment:
I think I'd rather have this look more like `LogPatternToVersion(versions_to_patterns=...)`. You should get the class name from `self.__class__` rather than hardcoding it.